cli/wallet/validator.go

```diff
@@ -204,3 +220,65 @@ func getDecryptedAccount(ctx *cli.Context, wall *wallet.Wallet, addr util.Uint16
 	}
 	return acc, nil
 }
+
+func getAccountState(ctx *cli.Context) error {
+	addrFlag := ctx.Generic("address").(*flags.Address)
+	if !addrFlag.IsSet {
+		return cli.NewExitError("address was not provided", 1)
+	}
+
+	gctx, cancel := options.GetTimeoutContext(ctx)
+	defer cancel()
+	c, exitErr := options.GetRPCClient(gctx, ctx)
+	if exitErr != nil {
+		return exitErr
+	}
+
+	neoHash, err := c.GetNativeContractHash(nativenames.Neo)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to get NEO contract hash: %w", err), 1)
+	}
+	res, err := c.InvokeFunction(neoHash, "getAccountState", []smartcontract.Parameter{
+		{
+			Type:  smartcontract.Hash160Type,
+			Value: addrFlag.Uint160(),
+		},
+	}, nil)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	if res.State != "HALT" {
+		return cli.NewExitError(fmt.Errorf("invocation failed: %s", res.FaultException), 1)
+	}
+	if len(res.Stack) == 0 {
+		return cli.NewExitError("result stack is empty", 1)
+	}
+	arr, ok := res.Stack[0].Value().([]stackitem.Item)
+	if !ok || len(arr) < 3 {
+		return cli.NewExitError("invalid result stackitem item", 1)
+	}
+	amount, err := arr[0].TryInteger()
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("invalid amount item: %w", err), 1)
+	}
+	dec, err := c.NEP17Decimals(neoHash)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to get decimals: %w", err), 1)
+	}
+	block, err := arr[1].TryInteger()
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("invalid block item: %w", err), 1)
+	}
+	pubBytes, err := arr[2].TryBytes()
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("invalid publick key item: %w", err), 1)
+	}
+	pk, err := keys.NewPublicKeyFromBytes(pubBytes, elliptic.P256())
```

Comment:
Will it work if there is no vote on this account?

Reply:
Yeah it can be `null`. I think we may want to make `NEOBalanceState` has `fromStackItem` we may want to extend it (remove any panics) and use it here.

Reply:
Fixed.